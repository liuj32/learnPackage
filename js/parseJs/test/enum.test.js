// enum.test.js
import enumFn from '../src/enum.js'

test('enum', () => {
  // 我们定义了俩个 action type
  const actionTypes = {
    ADD_TODO: 'add_todo',
    UPDATE_TODO: 'update_todo'
  }

  const safeActionTypes = enumFn(actionTypes)

  // 当读取一个不存在的枚举值时会报错
  // 因为 'DELETE_TODO' 并没有定义，所以此时会报错
  expect(() => {
    safeActionTypes['DELETE_TODO']
  }).toThrowErrorMatchingSnapshot()

  // 当删除一个枚举值时会报错
  expect(() => {
    delete safeActionTypes['ADD_TODO']
  }).toThrowErrorMatchingSnapshot()
})