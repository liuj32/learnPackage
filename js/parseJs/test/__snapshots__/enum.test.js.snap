// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enum 1`] = `"Unknown enum 'DELETE_TODO'"`;

exports[`enum 2`] = `"Enum is readonly"`;
